contracts/integrations/contracts/dydx/ZeroExV3ExchangeWrapper.sol

```diff
@@ -0,0 +1,329 @@
+/*
+
+    Copyright 2018 dYdX Trading Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+*/
+
+pragma solidity 0.5.9;
+pragma experimental ABIEncoderV2;
+
+import { SafeMath } from "./libs/SafeMath.sol";
+import { IExchange } from "@0x/contracts-exchange/contracts/src/interfaces/IExchange.sol";
+import { LibFillResults } from "@0x/contracts-exchange-libs/contracts/src/LibFillResults.sol";
+import { LibOrder } from "@0x/contracts-exchange-libs/contracts/src/LibOrder.sol";
+import { ExchangeReader } from "./interfaces/ExchangeReader.sol";
+import { ExchangeWrapper } from "./interfaces/ExchangeWrapper.sol";
+import { AdvancedTokenInteract } from "./libs/AdvancedTokenInteract.sol";
+import { MathHelpers } from "./libs/MathHelpers.sol";
+import { TokenInteract } from "./libs/TokenInteract.sol";
+
+
+/**
+ * @title ZeroExV3ExchangeWrapper
+ * @author dYdX/0x
+ *
+ * dYdX ExchangeWrapper to interface with 0x Version 3
+ */
+contract ZeroExV3ExchangeWrapper is
+    ExchangeWrapper,
+    ExchangeReader
+{
+    using SafeMath for uint256;
+    using TokenInteract for address;
+    using AdvancedTokenInteract for address;
+
+    // ============ Constants ============
+
+    // maximum gas price, 30 Gwei (choose value for deployment)
+    uint256 constant MAX_GAS_PRICE = 30000000000 wei;
```

Comment:
I'm wondering how necessary this is. It could be pretty limiting in times of congestions since it can't be upgraded and feels like it doesn't actually help that much if people want to make many trades with smaller gas prices.

If we do leave it in, I think it has to be upgradable.

Reply:
Yeah, I was considering making it upgradable. I lean towards leaving it in, but we can discuss offline